packages/api/controllers/project.js

```diff
@@ -3,17 +3,15 @@ import { generateId } from '../utils/id'
 import { log } from '../utils/logger'
 
 export async function createProject({
+  userId,
   programId,
   project,
 }) {
-  if (!programId) throw new Error('programId is required')
-  if (!project) throw new Error('project is required')
-
   const id = generateId()
   const item = {
     ...project,
+    programId: `${userId}#${programId}`,
```

Comment:
- should we call this the partition key explicitly so we don't assume this is the real `programId` - same comment applies to other places in this doc.
- Can we create a method to return the partition key so we keep it DRY

Reply:
Discussed on call. We'll call it `pk` and also call `id` `sk`